app/models/alert.rb

```diff
@@ -0,0 +1,62 @@
+# == Schema Information
+#
+# Table name: alerts
+#
+#  alert_type_id :bigint(8)
+#  created_at    :datetime         not null
+#  data          :json
+#  id            :bigint(8)        not null, primary key
+#  run_on        :date
+#  school_id     :bigint(8)
+#  status        :integer
+#  summary       :text
+#  updated_at    :datetime         not null
+#
+# Indexes
+#
+#  index_alerts_on_alert_type_id  (alert_type_id)
+#  index_alerts_on_school_id      (school_id)
+#
+
+class Alert < ApplicationRecord
+  belongs_to :school,     inverse_of: :alerts
+  belongs_to :alert_type, inverse_of: :alerts
+
+  delegate :title, to: :alert_type
+  delegate :description, to: :alert_type
+  delegate :display_fuel_type, to: :alert_type
+
+  scope :electricity,         -> { joins(:alert_type).merge(AlertType.electricity) }
+  scope :gas,                 -> { joins(:alert_type).merge(AlertType.gas) }
+  scope :no_fuel,             -> { joins(:alert_type).merge(AlertType.no_fuel) }
+  scope :termly,              -> { joins(:alert_type).merge(AlertType.termly) }
+  scope :weekly,              -> { joins(:alert_type).merge(AlertType.weekly) }
+  scope :before_each_holiday, -> { joins(:alert_type).merge(AlertType.before_each_holiday) }
+  scope :usable,              -> { where(status: [:good, :poor])}
+
+  enum status: [:good, :poor, :not_enough_data, :failed]
+
+  scope :latest, -> { order(created_at: :desc).group_by { |alert| [alert.alert_type_id] }.values.map(&:first) }
```

Comment:
This might get inefficient when we have lots of alerts. Could we group in the DB?